### ISSUE TITLE
Corrupted File! Failed to parse mod file | No such file or directory

### ISSUE THREAD

**Issue by kapitanluffy** [Senth/minecraft-mod-manager#179]
### Describe the bug 
Tried updating all mods and it showed errors on several mods

```
Failed to parse mod file mods\CraftTweaker-fabric-1.18.2-9.1.127.jar
Traceback (most recent call last):
  File "C:\Users\redacted\AppData\Roaming\Python\Python310\site-packages\minecraft_mod_manager\gateways\jar_parser.py", line 50, in _get_mod_info
    with ZipFile(file, "r") as zip:
  File "C:\Users\redacted\AppData\Local\Programs\Python\Python310\lib\zipfile.py", line 1240, in __init__
    self.fp = io.open(file, filemode)
FileNotFoundError: [Errno 2] No such file or directory: 'mods\\CraftTweaker-fabric-1.18.2-9.1.127.jar'

!!! Please report this and paste the above message !!!
    ❌ Corrupted file.
```

### Steps to reproduce
1. Ran `minecraft-mod-manager.exe --pretend -d mods -v 1.18.2 --mod-loader=fabric update`

### Expected Behavior
Will show mods to be updated

### Info
 - OS: win10
 - Python --version: 3.10
 - minecraft-mod-manager --version: 1.27

**Comment by Senth** [comment]
Thanks for reporting @kapitanluffy 🙂 

I think this is a duplicate of #93 since `--pretends` tries to read the version from the file that was downloaded.

I down-prioritized #93 since there were no votes or comments on it. But I will prioritize and fix this to v1.3 🙂 

**Background of bug**
I started using the version from the file instead of metadata from Curse/Modrinth since the metadata version number doesn't always match from Curse/Modrinth to that in the file (weirdly enough).

I'll also make sure to add a test or two for `--pretend` to make sure it continues to work correctly.

**Comment by Senth** [comment]
#93 has now been fixed in #183 and will be released in a few minutes in v1.3 🙂
